main/src/library/Iterator.flix

```diff
@@ -60,6 +60,27 @@ namespace Iterator {
         };
         Iterator(done, next)
 
+    ///
+    /// Returns an iterator that unfolds `f` until the resource at `x` is null.
+    ///
+    /// Assumes that `f` eventually exhausts the resource at `x`.
+    ///
+    /// Assumes `f` may throw a Java `RuntimeException`.
+    ///
+    pub def ofException(f: Unit -> a & Impure, x: Ref[a]): Iterator[Result[a, String]] =
```

Comment:
f should be effect polymorphic (same as below). (We want this to work both for pure and impure functions).